docs/nx/azure-react/generator-app-insights-web.md

```diff
@@ -0,0 +1,102 @@
+nx generate @ensono-stacks/azure-react:app-insights-web --name=<app-name> --connectionString=<KEY_NAME> --directory=<directory where the lib is placed> --importPath=<import path for the library> --tags=<tags to the library>
+```
+
+### Command line arguments
+
+The following command line arguments are available:
+
+| Option            | Description                                               | Type      | Required |
+| ---               | --------------------------------------------------------- | ---       | --- |
+| --name         | Library name                                      | string    | Required |
+| --connectionString  | The env variable for the connection string.        | string    | Required |
+| --directory          | A directory where the lib is placed.          | string    | Optional |
+| --importPath          | What import path would you like to use for the library?.          | string    | Optional |
+| --tags          | Add tags to the library (used for linting).          | string    | Optional |
+
+### Generator Output
+
+- Adds an app insights config file
+```json title="/src/app-insights-config.ts"
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {
+  DistributedTracingModes,
+  IConfig,
+} from '@microsoft/applicationinsights-common';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import type { IConfiguration } from '@microsoft/applicationinsights-core-js';
+
+const appInsightConfig: IConfiguration & IConfig = {
+  enableAutoRouteTracking: false,
+  enableCorsCorrelation: true,
+  enableRequestHeaderTracking: true,
+  enableResponseHeaderTracking: true,
+  disableFetchTracking: false,
+  distributedTracingMode: DistributedTracingModes.AI_AND_W3C,
+  enableAjaxPerfTracking: true
+};
+
+export default appInsightConfig;
+```
+
+- Adds a telemetry provider 
+```json title="/src/telemetry-provider.tsx"
+import {
+  ReactPlugin,
+  AppInsightsContext,
+} from '@microsoft/applicationinsights-react-js';
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { FC, ReactNode } from 'react';
+
+import appInsightConfig from './app-insights-config';
+
+const reactPlugin = new ReactPlugin();
+const connectionString = process.env.<%= connectionString %>;
+
+export const appInsights = new ApplicationInsights({
+  config: {
+    connectionString,
+    ...appInsightConfig,
+    extensions: [reactPlugin],
+  },
+});
+
+if (!appInsights.appInsights.isInitialized()) {
+  appInsights.loadAppInsights();
+}
+
+export const TelemetryProvider: FC<{ children?: ReactNode }> = ({ children }) => (
+  <AppInsightsContext.Provider value={reactPlugin}>
+    {children}
+  </AppInsightsContext.Provider>
+);
+```
+
+- Installs and adds microsoft applicationinsights packages to package.json
+```json title="/package.json"
+  "dependencies": {
+    ...OtherDependencies
+    "@microsoft/applicationinsights-react-js": "3.4.0",
+    "@microsoft/applicationinsights-web": "2.8.9",
+  }
+}
+```
+
```

Comment:
Is this the last step for the user? Do they not need to do any additional configuration? Do they need to import it to an app somewhere or is everything automated?

Reply:
They will need to import the provider and wrap their app with it. Added some additional docs to explain this and a link to the official MS docs for more info